src/core/segment_reader.rs

```diff
@@ -393,3 +393,57 @@ impl<'a> Iterator for SegmentReaderAliveDocsIterator<'a> {
         result
     }
 }
+
+#[cfg(test)]
+mod test {
+    use core::{Index, SegmentReader};
+    use schema::*;
+
+    #[test]
+    fn test_iterator() {
+        let mut schema_builder = SchemaBuilder::new();
+        schema_builder.add_text_field("name", TEXT);
+        let schema = schema_builder.build();
+        let index = Index::create_in_ram(schema.clone());
+        let name = schema.get_field("name").unwrap();
+
+        {
+            let mut index_writer = index.writer(50_000_000).unwrap();
+            let mut doc1 = Document::default();
+            doc1.add_text(name, "tantivy");
+            index_writer.add_document(doc1);
+
+            let mut doc2 = Document::default();
+            doc2.add_text(name, "horse");
+            index_writer.add_document(doc2);
+
+            // we should now have one segment with two docs
+            index_writer.commit().unwrap();
+        }
+
+        {
+            let mut index_writer2 = index.writer(50_000_000).unwrap();
+            index_writer2.delete_term(Term::from_field_text(name, "horse"));
+
+            // ok, now we should have a deleted doc
+            index_writer2.commit().unwrap();
+        }
+
+        let segments = index.searchable_segments().unwrap();
+        assert_eq!(segments.len(), 2);
+
+        // ok, we have two segments, lets grab the first segment and walk it
+        let sr = SegmentReader::open(&segments[0]).unwrap();
```

Comment:
I'd prefer you avoid calling `SegmentReader::open` and use the user path instead.

Call `.load_searcher()`, then acquire a searcher using
`index.searcher()`.

The readers are then accessible via  `searcher.segment_reader(0)` and `searcher.segment_reader(1)`.